base/compiler/optimize.jl

```diff
@@ -534,18 +534,30 @@ function ipo_escape_cache(mi_cache::MICache) where MICache
 end
 null_escape_cache(linfo::Union{InferenceResult,MethodInstance}) = nothing
 
-function run_passes(ci::CodeInfo, sv::OptimizationState, caller::InferenceResult)
-    @timeit "convert"   ir = convert_to_ircode(ci, sv)
-    @timeit "slot2reg"  ir = slot2reg(ir, ci, sv)
+macro pass(name, expr)
+    optimize_level = esc(:optimize_level)
+    stage = esc(:__stage__)
+    macrocall = :(@timeit $(esc(name)) $(esc(expr)))
+    macrocall.args[2] = __source__  # `@timeit` may want to use it
+    quote
+        $macrocall
+        $optimize_level < ($stage += 1) && return $(esc(:ir))
```

Comment:
How about this? So that we can specify the optimizer phase more directly:
```suggestion
        if $optimize_level isa String
            $optimize_level == $(esc(name)) && return $(esc(:ir))
        else
            $optimize_level < ($stage += 1) && return $(esc(:ir))
        end
```

Reply:
Good idea! I added it in ae3679d8114f47c9602ff7bad2107bb675f9f199